controllers/progressiverollout_controller_test.go

```diff
@@ -22,6 +25,26 @@ const (
 	interval = time.Millisecond * 10
 )
 
+type mockArgoCDAppClientCounter struct {
+	appsSynced []string
+	m          sync.Mutex
```

Comment:
The `mutex` is needed because the mocked `Sync` access the `appsSynced` to append stuff and there is the chance of a data race with the `Eventually()` in the test using this.